hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -826,26 +834,41 @@ update env ledger st ev = case (st, ev) of
     -- TODO: Is it really intuitive that we respond from the confirmed ledger if
     -- transactions are validated against the seen ledger?
     Effects [ClientEffect . ServerOutput.GetUTxOResponse headId $ getField @"utxo" $ getSnapshot confirmedSnapshot]
-  (Open openState, NetworkEvent _ _ (ReqDec{transaction})) ->
-    onOpenNetworkReqDec openState transaction
   (Open OpenState{headId, coordinatedHeadState, currentSlot}, ClientEvent Decommit{decommitTx}) -> do
-    -- TODO: Spec: require U̅ ◦ decTx /= ⊥
-    requireValidDecommitTx $ \utxoToDecommit ->
-      Effects
-        [ ClientEffect ServerOutput.DecommitRequested{headId, utxoToDecommit}
-        , NetworkEffect ReqDec{transaction = decommitTx}
-        ]
+    case mExistingDecommitTx of
+      Just existingDecommitTx ->
+        Effects
+          [ ClientEffect
+              ServerOutput.DecommitInvalid
+                { headId
+                , decommitInvalidReason = ServerOutput.DecommitAlreadyInFlight{decommitTx = existingDecommitTx}
+                }
+          ]
+      Nothing ->
+        requireValidDecommitTx headId ledger currentSlot coordinatedHeadState decommitTx $
+          Effects
+            [ NetworkEffect ReqDec{transaction = decommitTx, decommitRequester = party}
+            ]
    where
-    -- TODO: Spec: require U̅ ◦ decTx /= ⊥
```

Comment:
The TODO was a reminder that we need to align this with the spec. But we need to do that anyways for all things here, so fine.